docs/framework/interop/default-marshaling-for-strings.md

```diff
@@ -1,264 +1,267 @@
+    HWND hWnd,        // Handle to window or control.
+    LPTStr lpString,  // Text buffer.
+    int nMaxCount     // Maximum number of characters to copy.
 );
 ```
 
-<a name="cpcondefaultmarshalingforstringsanchor5"></a>
-
-## Strings Used in Platform Invoke  
- Platform invoke copies string arguments, converting from the .NET Framework format (Unicode) to the platform unmanaged format. Strings are immutable and are not copied back from unmanaged memory to managed memory when the call returns.  
-  
- The following table lists the marshaling options for strings when marshaled as a method argument of a platform invoke call. The <xref:System.Runtime.InteropServices.MarshalAsAttribute> attribute provides several <xref:System.Runtime.InteropServices.UnmanagedType> enumeration values to marshal strings.  
-  
-|Enumeration type|Description of unmanaged format|  
-|----------------------|-------------------------------------|  
-|`UnmanagedType.AnsiBStr`|A COM-style `BSTR` with a prefixed length and ANSI characters.|  
-|`UnmanagedType.BStr`|A COM-style `BSTR` with a prefixed length and Unicode characters.|  
-|`UnmanagedType.LPStr`|A pointer to a null-terminated array of ANSI characters.|  
-|`UnmanagedType.LPTStr`|A pointer to a null-terminated array of platform-dependent characters.|  
-|`UnmanagedType.LPWStr`|A pointer to a null-terminated array of Unicode characters.|  
-|`UnmanagedType.TBStr`|A COM-style `BSTR` with a prefixed length and platform-dependent characters.|  
-|`VBByRefStr`|A value that enables Visual Basic .NET to change a string in unmanaged code, and have the results reflected in managed code. This value is supported only for platform invoke. This is default value in Visual Basic for `ByVal` strings.|  
-  
- This table applies to strings. However, for <xref:System.Text.StringBuilder>, the only options allowed are `LPStr`, `LPTStr`, and `LPWStr`.  
-  
- The following type definition shows the correct use of `MarshalAsAttribute` for platform invoke calls.  
-  
-```vb  
-Class StringLibAPI      
-Public Declare Auto Sub PassLPStr Lib "StringLib.Dll" _  
-(<MarshalAs(UnmanagedType.LPStr)> s As String)      
-Public Declare Auto Sub PassLPWStr Lib "StringLib.Dll" _  
-(<MarshalAs(UnmanagedType.LPWStr)> s As String)      
-Public Declare Auto Sub PassLPTStr Lib "StringLib.Dll" _  
-(<MarshalAs(UnmanagedType.LPTStr)> s As String)      
-Public Declare Auto Sub PassBStr Lib "StringLib.Dll" _  
-(<MarshalAs(UnmanagedType.BStr)> s As String)      
-Public Declare Auto Sub PassAnsiBStr Lib "StringLib.Dll" _  
-(<MarshalAs(UnmanagedType.AnsiBStr)> s As String)      
-Public Declare Auto Sub PassTBStr Lib "StringLib.Dll" _  
-(<MarshalAs(UnmanagedType.TBStr)> s As String)  
-End Class  
+A `StringBuilder` can be dereferenced and modified by the callee, provided it does not exceed the capacity of the `StringBuilder`. The following code example demonstrates how `StringBuilder` can be initialized to a fixed length.
+
+```vb
```

Comment:
- There's no good reason here and in the C# example to define GetWindowText in a separate class. That just adds overhead to the example.

Reply:
The compiler asked to [Move P/Invokes to NativeMethods class](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1060-move-p-invokes-to-nativemethods-class) before, but I don't know when this warning disappeared.

Reply:
I didn't realize that. Thanks for pointing it out. That suggests that perhaps we should rename the class names to NativeMethods. If not, we should rename Win32API to WindowsAPI.